internal/config.go

```diff
@@ -26,6 +28,43 @@ type ConfigCORS struct {
 	AllowCredentials bool     `yaml:"allow_credentials"`
 }
 
+// ConfigProxy is a representation of section proxy of the yaml
+type ConfigProxy struct {
+	Url  string    `yaml:"url"`
+	Mode ProxyMode `yaml:"mode"`
+}
+
+// ProxyMode is enumeration of proxy server modes
+type ProxyMode int
+
+const (
+	// ProxyNone server is off
+	ProxyNone ProxyMode = iota
+	// ProxyMissing handle only missing requests are proxied
+	ProxyMissing
+	// ProxyAll all requests are proxied
+	ProxyAll
+)
+
+// UnmarshalYAML implementation of yaml.Unmarshaler interface
+func (mode *ProxyMode) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
you cand add somthing like this:

```suggestion
func proxyModeParseString(t string) (ProxyMode, error) {
	m := map[string]ProxyMode{
		"none":    ProxyNone,
		"missing": ProxyMissing,
		"all":     ProxyAll,
	}

	p, ok := m[t]
	if !ok {
		return ProxyNone, fmt.Errorf("unknown proxy mode: %s", t)
	}

	return p, nil
}
```

with this suggestion to could avoid the switch and only do 
```go
...
return proxyModeParseString(textMode)
```

Reply:
What are the cons in using map here? Maybe it is a bit shorter/

Reply:
Sorry, I don't understand your question... In your commit I don't see any map, the map is my suggestion, could you explain me a little bit more? thanks :)